### PR TITLE
chore(storybook): add links to source code

### PR DESCRIPTION
**Pull request by sunRock98**
[FX-NNNN]

### Description

Add links to source code of component so that user can easily navigate to implementation of component.

### Screenshots

Screenshot 2022-08-22 at 18 03 23

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -112,6 +112,11 @@ module.exports = {
       new IgnoreNotFoundPlugin(['OverridableComponent', 'BaseProps'])
     )
 
+    config.node = {
+      __filename: true,
+      __dirname: true,
+    }
```
Comment (dmaklygin) [inline .storybook/main.js:118]: could you please explain what it is needed for?

File: .storybook/components/PicassoBook/Book.js [hunk 2]
```diff
@@ -41,14 +41,26 @@ in https://github.com/toptal/picasso/blob/dcc2b12a05ebec7cbce2cd591149d4f056762a
...
   createBaseDocsLink = link => {
     return `[BASE documentation](${link})`
   }
 
+  createSourceLink = link => {
+    const linkWithoutStoryPath = link.split('/story/')[0]
+
+    const linkSplittedByHash = linkWithoutStoryPath.split('/')
+
+    const componentName = `${linkSplittedByHash[linkSplittedByHash.length - 1]}`
```
Comment (dmaklygin) [inline .storybook/components/PicassoBook/Book.js:57]: seems, quotes are not needed here

**Comment by dmaklygin** [comment]
@sunRock98 please, update this [DOC](https://github.com/toptal/picasso/blob/96fdc2c5fb6b623911fdf58bb95acf8682f8aa90/docs/contribution/creating-examples.md) as well

File: .storybook/main.js (hunk 1 above)
Comment (sunRock98) [reply .storybook/main.js:118]: The following Node.js options configure whether to use the filename of the input file relative to the context option.

https://webpack.js.org/configuration/node/#node__filename

Do you think I need to add a comment line for future development?

**Review by dmaklygin — APPROVED** [review]
I think, it's quite difficult to automatically determine the link to source code for EACH story.
I'm fine with this solution

**Review by denieler — COMMENTED** [review]
Code looks good 👍 But generally I don't think it's a good idea to add the source code link for Picasso components. If Picasso users need to go to the source code when using Picasso components - it means something is missed in the documentation. We would never know what exactly is missed if people would have always a "shortcut" to go to the source code and check by themselves

**Comment by sunRock98** [comment]
> Code looks good 👍 But generally I don't think it's a good idea to add the source code link for Picasso components. If Picasso users need to go to the source code when using Picasso components - it means something is missed in the documentation. We would never know what exactly is missed if people would have always a "shortcut" to go to the source code and check by themselves

I understand your skepticism. I would be ready to agree with you if it were some kind of portal for ordinary users. But our users are engineers. Thinking engineers. Who will not immediately run to your channel for advice as soon as they have a question. They will first of all try to figure it out themselves, and only as the last stronghold they will ask your team for help. At least that's what I do all the time. Therefore, it seems to me that we can give convenience for research here, and inspiration to study and figure it out on their own, which will reduce the burden on you and give you extra time to make and add new cool features. Butof coursee this is just my view and I do not insist.

**Comment by denieler** [comment]
> > Code looks good 👍 But generally I don't think it's a good idea to add the source code link for Picasso components. If Picasso users need to go to the source code when using Picasso components - it means something is missed in the documentation. We would never know what exactly is missed if people would have always a "shortcut" to go to the source code and check by themselves
> 
> @denieler I understand your skepticism. I would be ready to agree with you if it were some kind of portal for ordinary users. But our users are engineers. Thinking engineers. Who will not immediately run to your channel for advice as soon as they have a question. They will first of all try to figure it out themselves, and only as the last stronghold they will ask your team for help. At least that's what I do all the time. Therefore, it seems to me that we can give convenience for research here, and inspiration to study and figure it out on their own, which will reduce the burden on you and give you extra time to make and add new cool features. Butof coursee this is just my view and I do not insist.

I completely understand your point and also don't insist 😃 I did that also many times myself. Still, I think if I need to go to the source code - the docs are bad 😃

**Review by OndrejTuma — APPROVED** [review]
Code looks good to me 👍  thanks for the contribution. Although I agree with Daniel, this will improve things still 🙂

**Comment by dmaklygin** [comment]
@sunRock98 could you please update the changeset as well

**Review by augustobmoura — APPROVED** [review]
Code looks good! I find it very useful to have a link to the correct source code. Power users approve it!

**Pull request merged**